### PR TITLE
Update troubleshoot with more content

### PR DESCRIPTION
**Pull request by stevhliu**
This PR follows up on #16136 to include an `attention_mask` if user outputs are incorrect as a result of not masking the padding tokens. cc @patrickvonplaten @sgugger

### CODE REVIEW

File: docs/source/troubleshooting.mdx [hunk 1]
```diff
@@ -120,4 +120,28 @@ Another option is to get a better traceback from the GPU. Add the following envi
...
+
+```py
+>>> padding_id = 100
+>>> input_ids = torch.tensor([[7592, 2057, 2097, 2393, 9611, 2115], [7592, 100, 100, 100, 100, 100]])
+>>> outputs = model(input_ids)
+```
+
+The output value here will be different than if you included an attention mask because the model also attends to the padding tokens.
+
+Most of the time, you should provide an `attention_mask` to your model to ignore the padding tokens:
```
Comment (sgugger) [inline docs/source/troubleshooting.mdx:137]: You should mention it is automatically created by the tokenizer.

**Review by sgugger — APPROVED** [review]
Thanks for adding this here!

File: docs/source/troubleshooting.mdx [hunk 2]
```diff
@@ -120,4 +120,28 @@ Another option is to get a better traceback from the GPU. Add the following envi
...
 >>> os.environ["CUDA_LAUNCH_BLOCKING"] = "1"
-```
\ No newline at end of file
+```
+
+## Incorrect output when padding tokens aren't masked
+
+In some cases, you may notice the output `hidden_state` may be incorrect if the `input_ids` include padding tokens. For example, a batch of tensors may look like this before you pass it to `forward()`:
+
+```py
```
Comment (patrickvonplaten) [inline docs/source/troubleshooting.mdx:129]: Should we not load a model here so that the code can be executed? `model = BertModel.from_pretained("base-base-cased")`

File: docs/source/troubleshooting.mdx [hunk 3]
```diff
@@ -120,4 +120,28 @@ Another option is to get a better traceback from the GPU. Add the following envi
...
+>>> padding_id = 100
+>>> input_ids = torch.tensor([[7592, 2057, 2097, 2393, 9611, 2115], [7592, 100, 100, 100, 100, 100]])
```
Comment (patrickvonplaten) [inline docs/source/troubleshooting.mdx:131]:
```suggestion
>>> input_ids = torch.tensor([[7592, 2057, 2097, 2393, 9611, 2115], [7592, 0, 0, 0, 0, 0]])
```

**Review by patrickvonplaten — APPROVED** [review]
